evaporate.js

```diff
@@ -1847,10 +1800,46 @@
         return code.length ? {code: code, msg: msg} : {};
     }
 
+    var EvaporateHistoryCache = function () {
```

Comment:
This might be a good oportunity to start separating classes to extra files instead of making this one file even bigger, what do you think?